waku/v2/node/wakuoptions.go

```diff
@@ -340,6 +345,18 @@ func WithPubSubTopics(topics []string) WakuNodeOption {
 	}
 }
 
+func WithShards(shards []uint16) WakuNodeOption {
```

Comment:
we can, i kept it separate as shards can be passed nil in case of auto sharding. 
also there is another function WithPubsubTopics which also affects the shards. 

i know the logic looks little ugly, but this was just something i did for another user of libgowaku (who will ultimately move to nwaku).